addons/sys_profile/profileSimulator_v2.fsm

```diff
@@ -403,7 +403,7 @@ class FSM
        "                            _waypoints = _waypoints - [objNull];" \n
        "                                        " \n
        "                            //Needs review of any variables in hashes" \n
-       "                            if ((typeName _statements == ""ARRAY"") && {call compile (_statements select 0)}) then {call compile (_statements select 1)};" \n
+       "                            if ((_statements isEqualType []) && {(_statements select 0) == "true"}) then {call compile (_statements select 1)};" \n
```

Comment:
I'm not sure whether statements is a SQF statement (which it sounds like) vs just the string "true". You might need compile here.

Reply:
Considering the if-construction I assumed that the 1st parameter of  statements is a boolean, the second would be whatever. Hence, didn't remove the call compile for the 2nd param.

Reply:
Both the first and second index of `_statements` are strings, I _think_ they both need to be `call compile`d  so that the statement condition (index 0) can be evaluated.

Reply:
Reverted back to call compile